nixos/modules/services/web-apps/docuseal.nix

```diff
@@ -0,0 +1,174 @@
+        name of a user created elsewhere.
+      '';
+      type = lib.types.str;
+      default = "docuseal";
+    };
+
+    group = lib.mkOption {
+      description = lib.mdDoc ''
+        Group under which docuseal runs.
+      '';
+      type = lib.types.str;
+      default = "docuseal";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.docuseal = {
+      description = "DocuSeal server";
+      wantedBy = [ "multi-user.target" ];
+      environment = env;
+      serviceConfig = {
+        Type = "simple";
+        ExecStartPre = pkgs.writeShellScript "docuseal-pre-script" ''
+          cat > /var/lib/docuseal/docuseal.env <<EOF
+          SECRET_KEY_BASE="$(cat ${cfg.secretKeyBaseFile})"
+          EOF
+        '';
+        ExecStart = "${cfg.package}/bin/rails server --pid=/var/lib/docuseal/docuseal.pids";
+        Restart = "always";
+        EnvironmentFile = [ "docuseal.env" ] ++ cfg.extraEnvFiles;
+        # Runtime directory and mode
+        RuntimeDirectory = "docuseal";
+        RuntimeDirectoryMode = "0750";
+        # System Call Filtering
+        SystemCallFilter = [ "@system-service" "~@privileged" ];
+        # User and group
+        User = cfg.user;
+        Group = cfg.group;
+        DynamicUser = true;
+        # Working directory
+        WorkingDirectory = cfg.package;
+        # State directory and mode
+        StateDirectory = "docuseal";
+        StateDirectoryMode = "0750";
+        # Logs directory and mode
+        LogsDirectory = "docuseal";
+        LogsDirectoryMode = "0750";
+        # Proc filesystem
+        ProcSubset = "pid";
+        ProtectProc = "invisible";
+        # Access write directories
+        UMask = "0027";
+        # Capabilities
+        CapabilityBoundingSet = "";
+        # Security
+        NoNewPrivileges = true;
+        # Sandboxing
+        ProtectSystem = "strict";
+        ProtectHome = true;
+        PrivateTmp = true;
+        PrivateDevices = true;
+        #PrivateUsers = true;
+        PrivateUsers = (cfg.port >= 1024);
+        ProtectClock = true;
+        ProtectHostname = true;
+        ProtectKernelLogs = true;
+        ProtectKernelModules = true;
+        ProtectKernelTunables = true;
+        ProtectControlGroups = true;
+        RestrictAddressFamilies = [ "AF_UNIX" "AF_INET" "AF_INET6" "AF_NETLINK" ];
+        RestrictNamespaces = true;
+        LockPersonality = true;
+        MemoryDenyWriteExecute = false;
+        RestrictRealtime = true;
+        RestrictSUIDSGID = true;
+        RemoveIPC = true;
+        PrivateMounts = true;
+        # System Call Filtering
+        SystemCallArchitectures = "native";
+      } // lib.optionalAttrs (cfg.port < 1024) {
+        AmbientCapabilities = [ "CAP_NET_BIND_SERVICE" ];
+        CapabilityBoundingSet = [ "CAP_NET_BIND_SERVICE" ];
+      };
+    };
+
+    systemd.tmpfiles.rules = [
+      "d '/var/lib/docuseal' - ${cfg.user} ${cfg.group} - -"
+    ];
```

Comment:
Same question as `user`/`group`.

The state directory is automatically created with the correct permissions by `systemd`, no need to use `tmpfiles`.